### PR TITLE
Add scheduler name support - [Update #990]

### PR DESCRIPTION
**Pull request by bbsbb**
Hey there,

This is an updated, rebased continuation of Micah Coletti's PR allowing specification of an alternative scheduler.

If there is additional work that needs to be done on this to meet a standard deserving of merging, I'm up for it.

Would really appreciate involvement and direct answer if this is something that could become part of the postgres operator as it is a blocking issue for us.

Thank you for the great work on the project!

### CODE REVIEW

**Comment by Jan-M** [comment]
:+1:

File: pkg/apis/acid.zalan.do/v1/crds.go [hunk 1]
```diff
@@ -569,6 +569,9 @@ var PostgresCRDResourceValidation = apiextv1.CustomResourceValidation{
 							},
 						},
 					},
+					"schedulerName": {
+						Type: "string",
+					},
```
Comment (FxKu) [inline pkg/apis/acid.zalan.do/v1/crds.go:574]: could you add this part also to the manifests for the Postgresql CRD? [Here](https://github.com/zalando/postgres-operator/blob/master/manifests/postgresql.crd.yaml) and [there](https://github.com/zalando/postgres-operator/blob/master/charts/postgres-operator/crds/postgresqls.yaml).

**Comment by FxKu** [comment]
should be documented here: https://github.com/zalando/postgres-operator/blob/master/docs/reference/cluster_manifest.md

File: pkg/apis/acid.zalan.do/v1/crds.go (hunk 1 above)
Comment (bbsbb) [reply pkg/apis/acid.zalan.do/v1/crds.go]: Done.
Also noticed that the custom resource defs follow alphabetical order and I hadn't respected that, so adjusted it as well.

**Comment by bbsbb** [comment]
CRDs adjusted, documentation for the spec option added.

File: docs/reference/cluster_manifest.md [hunk 2]
```diff
@@ -65,6 +65,10 @@ These parameters are grouped directly under  the `spec` key in the manifest.
   custom Docker image that overrides the **docker_image** operator parameter.
   It should be a [Spilo](https://github.com/zalando/spilo) image. Optional.
 
+* **schedulerName**
+  allows choice of scheduling profile for pods. If no value is provided
+  assumed is K8s default - `default-scheduler`. Optional.
```
Comment (FxKu) [inline docs/reference/cluster_manifest.md:70]:
```suggestion
  specifies the scheduling profile for pods. If no value is provided
  K8s' `default-scheduler` will be used. Optional.
```

**Comment by FxKu** [comment]
can you rebase with the master. I've just merged a fix for CRDs and also updated go modules. Ideally, only changes relevant for this PR should be contained (e.g. no change to makefile etc.).

**Comment by bbsbb** [comment]
> can you rebase with the master. I've just merged a fix for CRDs and also updated go modules. Ideally, only changes relevant for this PR should be contained (e.g. no change to makefile etc.).

On it.

**Comment by bbsbb** [comment]
- Removed Makefile EOL change & deps bumps for clean PR :x: 
- Adjusted documentation per suggestion. :heavy_check_mark: 
- Rebased to latest master & squashed :heavy_check_mark: 

Cheers, let me know if anything else is necessary.

File: docs/reference/cluster_manifest.md [hunk 3]
```diff
@@ -65,6 +65,10 @@ These parameters are grouped directly under  the `spec` key in the manifest.
   custom Docker image that overrides the **docker_image** operator parameter.
   It should be a [Spilo](https://github.com/zalando/spilo) image. Optional.
 
+* **schedulerName**
+  specifies the scheduling profile for pods. If no value is provided
```
Comment (FxKu) [inline docs/reference/cluster_manifest.md:69]:
```suggestion
  specifies the scheduling profile for database pods. If no value is provided
```

**Comment by FxKu** [comment]
Did one minor update as the scheduler name is only set for the database pods.

**Comment by FxKu** [comment]
:+1:

**Comment by Jan-M** [comment]
:+1:

**Comment by FxKu** [comment]
:+1:

**Pull request merged**